e2e/test-component/scenarios/embedding-sdk/guest-embed-parameters.cy.spec.tsx

```diff
@@ -0,0 +1,850 @@
+        cy.findByPlaceholderText("Search by Name or enter an ID").type("Hud");
+
+        cy.wait("@parameterSearchApiRequest");
+
+        cy.get("@parameterSearchApiRequest.all").then((interceptions) => {
+          expect(interceptions).to.have.length.greaterThan(0);
+        });
+
+        // Verify search results are displayed
+        cy.findByText("Hudson Borer").should("exist");
+      });
+    });
+
+    it("should show remapped parameter values and trigger proper `/embed remapping endpoint", () => {
+      signInAsAdminAndSetupGuestEmbedding({
+        token: "starter",
+      });
+
+      // Set up internal remapping for ORDERS.QUANTITY
+      cy.request("POST", `/api/field/${ORDERS.QUANTITY}/dimension`, {
+        name: "Quantity",
+        type: "internal",
+        human_readable_field_id: null,
+      });
+
+      cy.request("GET", `/api/field/${ORDERS.QUANTITY}/values`).then(
+        ({ body }: { body: { values: [number][] } }) => {
+          cy.request("POST", `/api/field/${ORDERS.QUANTITY}/values`, {
+            values: body.values.map(([value]) => [value, `N${value}`]),
+          });
+        },
+      );
+
+      // Set up external remapping: ORDERS.PRODUCT_ID -> PRODUCTS.TITLE
+      cy.request("POST", `/api/field/${ORDERS.PRODUCT_ID}/dimension`, {
+        name: "Product ID",
+        type: "external",
+        human_readable_field_id: PRODUCTS.TITLE,
+      });
+
+      const dashboardTemplateTags: TemplateTags = {
+        quantity: {
+          id: "quantity",
+          name: "quantity",
+          "display-name": "Internal",
+          type: "dimension",
+          "widget-type": "number/=",
+          dimension: ["field", ORDERS.QUANTITY, null],
+        },
+        product_id_fk: {
+          id: "product_id_fk",
+          name: "product_id_fk",
+          "display-name": "FK",
+          type: "dimension",
+          "widget-type": "id",
+          dimension: ["field", ORDERS.PRODUCT_ID, null],
+        },
+        user_id_pk: {
+          id: "user_id_pk",
+          name: "user_id_pk",
+          "display-name": "PK->Name",
+          type: "dimension",
+          "widget-type": "id",
+          dimension: ["field", PEOPLE.ID, null],
+        },
+      };
+
+      const dashboardParameters = [
+        createMockParameter({
+          id: "quantity",
+          name: "Internal",
+          slug: "quantity",
+          type: "number/=",
+        }),
+        createMockParameter({
+          id: "product_id_fk",
+          name: "FK",
+          slug: "product_id_fk",
+          type: "id",
+        }),
+        createMockParameter({
+          id: "user_id_pk",
+          name: "PK->Name",
+          slug: "user_id_pk",
+          type: "id",
+        }),
+      ];
+
+      createNativeQuestionAndDashboard({
+        questionDetails: {
+          name: "Orders native question",
+          native: {
+            query:
+              "SELECT * " +
```

Comment:
Multiline string like this looks a bit weird. Would using backticks more suitable? The output would contain a lot of leading spaces if you put it in the same indent here, but spaces should be negligible if you don't have to assert the query anywhere.